lib/HTFeed/QueueRunner.pm

```diff
@@ -104,19 +106,30 @@ sub validate_job {
     return 0;
 
   }
-
 }
 
 sub run_job_sequence {
-  my $self = shift;
-  my $job = shift;
-
-  while($job){
-    get_logger()->info("next job: " . $job->namespace . "." . $job->id . " " . $job->stage_class);
-
-    $job->run_job($self->{clean});
-    $job = $job->successor;
-  }
+    my $self = shift;
+    my $job = shift;
+
+    while($job){
+	get_logger()->info(
+	    "next job: " . $job->namespace . "." . $job->id . " " . $job->stage_class
+	);
+	my $jobtype = (split("::", $job->stage_class))[-1];
+	my $t_start = Time::HiRes::time();
+	$job->run_job($self->{clean});
+	my $t_end = Time::HiRes::time();                  # gives milliseconds,
+	my $t_delta_seconds = ($t_end - $t_start) * 1000; # multiply to seconds
+	$self->{job_metrics}->add($jobtype."_seconds", $t_end - $t_start);
```

Comment:
Would it make sense to push this logic into `Job` rather than having it here? If we do keep it here I would suggest extracting it as a separate method though.

Reply:
mentioned in added issue. resolving convo